packages/icon/package.py

```diff
@@ -0,0 +1,247 @@
+    variant('host', default='daint', description='Build on host daint', multi=False)
+    variant('cuda_arch', default='60', description='Build with cuda_arch', values=('70', '60', '37'), multi=False)
+    variant('claw', default=True, description='Build with claw directories enabled')
+    variant('rte-rrtmgp', default=True, description='Build with rte-rrtmgp enabled')
+    variant('mpi-checks', default=False, description='Build with mpi-check enabled')
+    variant('openmp', default=True, description='Build with openmp enabled')
+    variant('serialize', default=False, description='Build with serialization enabled')
+    variant('eccodes', default=False, description='Build with grib2 enabled')
+    variant('test_name', default='none', description='Launch test: test_name after installation')
+    variant('skip-config', default=False, description='Skip configure phase')
+    variant('ham', default=True, description='Build with hammoz and atm_phy_echam enabled.')
+
+    conflicts('+openmp', when='%intel')
+    conflicts('+openmp', when='%cce')
+    conflicts('+claw', when='%intel')
+    conflicts('+claw', when='%cce')
+    conflicts('icon_target=gpu', when='%intel')
+    conflicts('icon_target=gpu', when='%cce')
+
+    atm_phy_echam_submodels_namelists_dir = 'externals/atm_phy_echam_submodels/namelists'
+
+    def setup_build_environment(self, env):
+        self.setup_run_environment(env)
+        if self.spec.variants['icon_target'].value == 'gpu':
+            env.set('CUDA_HOME', self.spec['cuda'].prefix)
+
+    @run_before('configure')
+    def generate_hammoz_nml(self):
+        if '@ham' in self.spec:
+            with working_dir(self.atm_phy_echam_submodels_namelists_dir):
+                make()
+
+    def configure_args(self):
+        args = []
+
+        CFLAGS=''
+        CPPFLAGS=''
+        FCFLAGS=''
+        LDFLAGS=''
+        LIBS=''
+
+        # Icon-hammoz:
+        if '+ham' in self.spec:
+            args.append('--enable-atm-phy-echam-submodels')
+            args.append('--enable-hammoz')
+
+        # Eccodes library:
+        if '+eccodes' in self.spec:
+            args.append('--enable-grib2')
+            args.append('--without-external-cdi')
+            ECCODESI='-I' + self.spec['eccodes'].prefix + '/include '
+            ECCODESL='-L' + self.spec['eccodes'].prefix + '/lib '
+            ECCODES_LIBS='-leccodes '
+            CPPFLAGS+=ECCODESI
+            LDFLAGS+=ECCODESL
+            LIBS+=ECCODES_LIBS
+
+        # Claw library
+        if '+claw' in self.spec:
+            args.append('CLAW=' + self.spec['claw'].prefix + '/bin/clawfc')
+            args.append('CLAWFLAGS=-I' + self.spec['netcdf-c'].prefix + '/' + self.compiler.name + '/' + str(self.compiler.version.up_to(2))  + '/include')
+            args.append('--enable-claw')
+
+        # Rte-rttmgp
+        if '+rte-rrtmgp' in self.spec:
+            args.append('--enable-rte-rrtmgp')
+
+        # Mpi-checks
+        if '~mpi-checks' in self.spec:
```

Comment:
I think these checks are turned off in most configure scripts by default (at least the ones for cscs). Here it looks like they are on by default?